LoopKitUI/Views/ExpandableDatePicker.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  ExpandableDatePicker.swift
+//  LoopKitUI
+//
+//  Created by Anna Quinlan on 8/12/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import SwiftUI
+
+public struct ExpandableDatePicker: View {
+    @State var dateShouldExpand = false
+    @Binding var date: Date
+    let placeholderText: String
+    let pickerRange: ClosedRange<Date>
+    @State public var userDidTap: Bool = false
+    
+    public init (
+        with date: Binding<Date>,
+        pickerRange: ClosedRange<Date>? = nil,
+        placeholderText: String = ""
+    ) {
+        _date = date
+        self.placeholderText = placeholderText
+        
+        let today = Date()
+        self.pickerRange = pickerRange ?? today.addingTimeInterval(-.hours(24))...today
+    }
+    
+    public var body: some View {
+        VStack(spacing: 0) {
+            HStack {
+                dateFieldText
+                Spacer()
+            }
+            .padding()
+            .frame(minWidth: 0, maxWidth: .infinity).onTapGesture {
+                self.userDidTap = true
+                self.dateShouldExpand.toggle()
+            }
+            if dateShouldExpand {
+                DatePicker("", selection: $date, in: pickerRange, displayedComponents: .date)
+                .labelsHidden()
+            }
+        }
+    }
+    
+    private var dateFieldText: some View {
+        if userDidTap {
+            return Text(dateFormatter.string(from: date))
+            // Show the placeholder text if user hasn't interacted with picker
+        } else {
+            return Text(placeholderText).foregroundColor(Color(UIColor.lightGray))
+        }
+    }
+    
+    private var dateFormatter: DateFormatter {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "MM/dd/yyyy"
```

Comment:
Is this format really needed? I would think we allow the user to define their preferred format and use that. What about replacing with `formatter.dateStyle = .medium`?

Reply:
Changed - that makes complete sense